src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
-def make_directory(path):
-    """
-    Makes a directory if it does not already exist.
-
-    :param path: path to output directory
-    """
-    try:
-        os.makedirs(path)
-    except OSError as exc:  # Python >2.5
-        if exc.errno == errno.EEXIST and os.path.isdir(path):
-            pass
-        else:
-            raise
-
-def download_url(job, url, filename):
-    """
-    Downloads a file from a URL and places it in the jobStore
-
-    :param job: Job instance
-    :param url: data url, str
-    :param filename: name given to downloaded data, str
-
-    :return: fileStore promise
-    """
-    work_dir = job.fileStore.getLocalTempDir()
-    file_path = os.path.join(work_dir, filename)
-    if not os.path.exists(file_path):
-        if url.startswith('s3:'):
-            download_from_s3_url(file_path, url)
-        else:
-            try:
-                subprocess.check_call(['curl', '-fs', '--retry', '5', '--create-dir', url, '-o', file_path])
-            except subprocess.CalledProcessError as cpe:
-                raise RuntimeError(
-                    '\nNecessary file could not be acquired: %s. Got error "%s". Check input URL' % (url, cpe))
-            except OSError:
-                raise RuntimeError('Failed to find "curl". Install via "apt-get install curl"')
-    assert os.path.exists(file_path)
-    return job.fileStore.writeGlobalFile(file_path)
-
-
-def return_input_paths(job, work_dir, ids, *filenames):
-    """
-    Given one or more filenames, reads the file from the fileStore and
-    copies it (or makes a hard link) into the working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory, str
-    :param ids: dictionary of shared file ids
-    :param filenames: remaining arguments are filenames
-
-    :returns: list of paths to files
+def download_and_run(job, uuid, url, config, rg_line=None):
     """
-    paths = OrderedDict()
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            file_path = job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-        else:
-            file_path = os.path.join(work_dir, filename)
-        paths[filename] = file_path
-
-    return paths.values()
-
+    Downloads reference files and runs the GATK best practices germline pipeline for a single
+    samples.
 
-def write_to_filestore(job, work_dir, ids, *filenames):
+    :param JobFunctionWrappingJob job: Toil Job instance
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to BAM file
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: None
     """
-    Given one or more file names in working directory, writes
-    files to filestore and stores the filestore promise in a dictionary
+    get_shared_files = job.wrapJobFn(download_shared_files, config).encapsulate()
```

Comment:
Can you open a ticket to refactor this later? Ideally, we should run this once for all samples, and then pass the filestore IDs around.